spec/models/pet_spec.rb

```diff
@@ -3,6 +3,8 @@
 RSpec.describe Pet, type: :model do
   describe "relationships" do
     it { should belong_to(:shelter) }
+    it { should have_many :pet_applications }
+    it { should have_many(:applications).through(:pet_applications) }
   end
 
```

Comment:
Make sure you test every method you add to the model at the unit level! you're missing some model method tests here